bin/build

```diff
@@ -65,15 +68,34 @@ findFile() {
   head -n 1
 }
 
-# Recursively search input directory for a package.json
+# Recursively search output directory for a package.json
 PACKAGE_PATH=$(findFile "package.json" $OUTPUT_DIR/bundle/programs/server)
+export PACKAGE_PATH
 
-if [[ $PACKAGE_PATH ]]; then
-  export PACKAGE_PATH
+NODE_VERSION=$(get-version --engine node $PACKAGE_PATH 2>/dev/null) || true
+NPM_VERSION=$(get-version --engine npm $PACKAGE_PATH 2>/dev/null) || true
 
-  if [[ $NPM_USER ]] && [[ $NPM_PASSWORD ]] && [[ $NPM_EMAIL ]]; then
-    npm-login
-  fi
+if [[ 'v$NODE_VERSION' != $(node -v) ]]; then
+  nvm install $NODE_VERSION > /dev/null 2>&1
+fi
 
-  npm-install
+if [[ $NPM_VERSION  != $(npm -v) ]]; then
+  npm install npm@$NPM_VERSION --global > /dev/null 2>&1
 fi
+
+nvm alias deploy $(nvm current)
+
+if [[ -f $INPUT_DIR/package.json ]]; then
+  printf "Installing root package.json dependencies.\n"
+  npm install --production
+fi
+
+if [[ -d $INPUT_DIR/node_modules ]]; then
+  mv $INPUT_DIR/node_modules $OUTPUT_DIR
```

Comment:
It's moved to after due to the switching of node versions (if need be). Once the node versions are switch, we no longer have access to demeteorizer unless we install it again. But it was still before the final `npm install`